### PR TITLE
fix: Transformers NVIDIA_VISIBLE_DEVICES value type casting

### PR DESCRIPTION
**Pull request by ssheng**
As reported in https://github.com/bentoml/BentoML/issues/2620, Transformers accepts an integer device ID. The current implementation pass the value of NVIDIA_VISIBLE_DEVICES environment variable as device without converting to an integer.

## Checklist:

- [ ] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by aarnphm — APPROVED** [review]
We need to add test for this. Maybe unit test under `tests/unit/_internal/frameworks`

File: bentoml/_internal/frameworks/transformers.py [hunk 1]
```diff
@@ -262,12 +262,17 @@ class TransformersRunnable(bentoml.Runnable):
         def __init__(self):
             super().__init__()
 
-            available_gpus = os.getenv("NVIDIA_VISIBLE_DEVICES")
+            available_gpus: str = os.getenv("NVIDIA_VISIBLE_DEVICES")
```
Comment (sauyon) [inline bentoml/_internal/frameworks/transformers.py:265]: Should we support `NVIDIA_VISIBLE_DEVICES=all` or `NVIDIA_VISIBLE_DEVICES=1,2`?

Comment (ssheng) [reply bentoml/_internal/frameworks/transformers.py:265]: My understanding is that breaking down `all` and `1,2` to individual devices is the job of the scheduling strategy.

File: bentoml/_internal/frameworks/FRAMEWORK_TEMPLATE_PY [hunk 2]
```diff
@@ -179,7 +179,7 @@ def get_runnable(bento_model: bentoml.Model) -> t.Type[bentoml.Runnable]:
                 # bento_model.info.options.*
             )
 
-            available_gpus = os.getenv("NVIDIA_VISIBLE_DEVICES")
```
Comment (aarnphm) [inline bentoml/_internal/frameworks/FRAMEWORK_TEMPLATE_PY:182]: https://gitlab.com/nvidia/container-images/cuda/-/blob/master/templates/ubuntu/base-dockerfile.j2#L96

**Review by aarnphm — APPROVED** [review]

**Pull request merged**